api/src/transactions.rs

```diff
@@ -508,99 +562,192 @@ impl TransactionsApi {
-                &ledger_info,
             )),
-            MempoolStatusCode::UnknownStatus => {
-                Err(SubmitTransactionError::service_unavailable_with_code(
-                    format!("Transaction was rejected with status {}", mempool_status),
-                    AptosErrorCode::InternalError,
-                    &ledger_info,
-                ))
-            }
+            MempoolStatusCode::UnknownStatus => Err(AptosError::new_with_error_code(
+                format!("Transaction was rejected with status {}", mempool_status,),
+                AptosErrorCode::InternalError,
+            )),
         }
     }
 
+    async fn create(
+        &self,
+        accept_type: &AcceptType,
+        ledger_info: &LedgerInfo,
+        txn: SignedTransaction,
+    ) -> SubmitTransactionResult<PendingTransaction> {
+        match self.create_internal(txn).await {
+            Ok(pending_txn) => match accept_type {
+                AcceptType::Json => SubmitTransactionResponse::try_from_json((
+                    pending_txn,
+                    ledger_info,
+                    SubmitTransactionResponseStatus::Accepted,
+                )),
+                AcceptType::Bcs => SubmitTransactionResponse::try_from_bcs((
+                    pending_txn.hash.0,
+                    ledger_info,
+                    SubmitTransactionResponseStatus::Accepted,
+                )),
+            },
+            Err(error) => match error.error_code {
+                AptosErrorCode::InternalError => Err(
+                    SubmitTransactionError::internal_from_aptos_error(error, ledger_info),
+                ),
+                AptosErrorCode::VmError
+                | AptosErrorCode::SequenceNumberTooOld
+                | AptosErrorCode::InvalidTransactionUpdate => Err(
+                    SubmitTransactionError::bad_request_from_aptos_error(error, ledger_info),
+                ),
+                AptosErrorCode::MempoolIsFull => Err(
+                    SubmitTransactionError::insufficient_storage_from_aptos_error(
+                        error,
+                        ledger_info,
+                    ),
+                ),
+                _ => Err(SubmitTransactionError::internal_from_aptos_error(
+                    error,
+                    ledger_info,
+                )),
+            },
+        }
+    }
+
+    async fn create_batch(
+        &self,
+        accept_type: &AcceptType,
+        ledger_info: &LedgerInfo,
+        txns: Vec<SignedTransaction>,
+    ) -> SubmitTransactionsBatchResult<SubmitTransactionsBatchExecutionResult> {
+        let mut txn_results = Vec::new();
+        for txn in txns {
+            txn_results.push(match self.create_internal(txn).await {
+                Ok(pending_txn) => {
+                    SubmitTransactionsBatchSingleExecutionResult::SUCCESS(pending_txn.hash)
+                }
+                Err(error) => SubmitTransactionsBatchSingleExecutionResult::FAILURE(error),
```

Comment:
The problem I see here, is I don't know which transactions failed, just the lack of which ones succeeded (and errors without associated txns).

Shoudl we add hash to the execution failure?

Reply:
They are returned in the same order as in the request. I do match them in account_minter, because I do need to know,  as I do retries there. 

But if you prefer hash to be added in the response as well, I can do that too.

But which hash are you referring to,  I see only hash for submitted transaction, not for UserTransactionRequestInner?

Reply:
Yes, it is the `UserTransactionRequestInner` hash.  I see though, it's in order so I guess it's fine as long as it's order isn't changed somehow.